make.js

```diff
@@ -176,6 +181,83 @@ CLI.gendocs = function() {
     banner('Generating docs successful', true);
 }
 
+//
+// Generate debugger configuration (currently only for node tasks)
+// ex: node make.js gendebugconfig --agent <path to agent directory>
+// ex: node make.js gendebugconfig --task <task> --agent <path to agent directory>
+//
+CLI.gendebugconfig = function() {
+    const agentPath = argv.agent;
```

Comment:
we author build config generation here (in C#): https://github.com/microsoft/azure-pipelines-tasks/tree/master/BuildConfigGen

It's better to add this there b/c it has the buildconfig definitions meta data (so new ones we have will have your changes) and correct code for calculating patch versions for multiple builds configs
 
Binaries are not checked in, so usage is similar e.g. dotnet BuildConfigGen

Simply open the sln in Visual Studio to work on it

Reply:
Would it be also possible to simply parse the version map file for the corresponding task?
Or do you mean supplying an optional parameter to the build command in here, in which case it would also generate the launch configurations?